pkg/agent/plugin/nodeattestor/aws/iid_test.go

```diff
@@ -0,0 +1,192 @@
+package aws
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/spiffe/spire/pkg/common/plugin/aws"
+	"github.com/spiffe/spire/proto/agent/nodeattestor"
+	"github.com/spiffe/spire/proto/common/plugin"
+	"github.com/stretchr/testify/suite"
+)
+
+func (s *Suite) buildDefaultIIDDocAndSig() (docBytes []byte, sigBytes []byte) {
+	// doc body
+	doc := aws.InstanceIdentityDocument{
+		AccountID:  "test-account",
+		InstanceID: "test-instance",
+		Region:     "test-region",
+	}
+	docBytes, err := json.Marshal(doc)
+	s.Require().NoError(err)
+
+	rng := rand.Reader
+	bitSize := 2048
+	key, err := rsa.GenerateKey(rng, bitSize)
+
```

Comment:
check error. also, unless you really need a new key for each usage, ideally the key is generated before hand to speed up the test execution. It can be loaded at test package init time [like so](https://github.com/spiffe/spire/blob/master/pkg/agent/endpoints/sds/handler_test.go#L70).